src/linker/Linker/MessageContainer.cs

```diff
@@ -132,6 +132,16 @@ private static MessageContainer CreateWarningMessageContainer (LinkContext conte
 			if (version > context.WarnVersion)
 				return Empty;
 
+			if (origin.MemberDefinition != null) {
+				var assembly = origin.MemberDefinition.DeclaringType.Module.Assembly;
+				var assemblyName = assembly.Name.Name;
+				if (context.AreWarningsCollapsed (assemblyName)) {
+					if (context.AssembliesWithGeneratedCollapsedWarnings.Add (assemblyName))
+						context.LogWarning ($"Assembly '{assemblyName}' produced trim warnings. For more information see https://aka.ms/dotnet-illink/libraries", 2104, context.GetAssemblyLocation (assembly));
```

Comment:
Is the sentence correct? The assembly itself didn't produce any warnings, right? Something like `Trimming warnings for assembly '{assemblyName} were suppressed.' would make more sense to me.

Reply:
How do you feel about "Assembly 'asm' _has_ trim warnings"?

I'd avoid "suppressed" since that has an existing specific meaning. It's true that the assembly isn't technically producing warnings (the analysis logic is) - but I also don't like the passive voice "Trimming warnings were produced...". And "Trim analysis produced warnings for assembly 'asm'" seems a bit wordy and puts the source of the warning at the end of the message. Other verbs we could use are "caused" or "originated".

I'm having trouble picking the best wording here. Maybe @agocke has suggestions.

Reply:
Oh, and regarding the suggestion in https://github.com/mono/linker/pull/1937#discussion_r606382945:
> The assembly 'assembly' is not trim-safe...

I am not sure if we want to talk about it in terms of "safety" since that has other connotations. IIRC @jkotas made this point in past discussions.